tests/scripts/parse_notice.py

```diff
@@ -0,0 +1,277 @@
+        self.scanned_json_file: str = scanned_json_file
+        self.processed_packages: dict[str, dict[str, str]] = {}
+        self.required_packages: dict[str, str] = {}
+        self.notice_file_name: str = "NOTICE.txt"
+        self.mode: str = cli_mode_argument
+
+        self.read_requirements()
+
+        scanned_results_data = self.read_content_from_file(self.scanned_json_file)
+
+        if not scanned_results_data:
+            raise ValueError(f"{self.scanned_json_file} is empty")
+
+        self.scanned_results_json: Any = json.loads(scanned_results_data)
+
+        notice_file_content: str = self.read_content_from_file("NOTICE.txt")
+
+        package_pattern = r"(?:Package: ([^\n]+))"
+        existing_packages: list[str] = re.findall(package_pattern, notice_file_content)
+        existing_packages.sort()
+
+        requirements_name_from_file = [requirement for requirement in self.required_packages.keys()]
+        requirements_name_from_file.sort()
+
+        real_requirements_name = [requirement for requirement in self.required_packages.values()]
+        real_requirements_name.sort()
+
+        if self.mode == "--check":
+            if real_requirements_name == existing_packages:
+                print("[!] There is no new package listed in the requirements files")
+                sys.exit()
+            else:
+                for new_package in requirements_name_from_file:
+                    if self.required_packages[new_package] not in existing_packages:
+                        real_package_name: str = self.required_packages[new_package]
+                        print(f"[!] New package found: '{real_package_name}'")
+
+                print("[!] Run the script with the 'fix' argument in order to add it to the NOTICE.txt file")
+                sys.exit(1)
+
+        elif self.mode == "--fix":
+            if real_requirements_name == existing_packages:
+                print("[!] There is no new package listed in the requirements files")
+                sys.exit()
+
+            for new_package in requirements_name_from_file:
+                if self.required_packages[new_package] not in existing_packages:
+                    real_package_name = self.required_packages[new_package]
+
+                    print(f"[!] New package found: '{real_package_name}'")
+                    self.process_package(required_package=new_package)
+                    self.verify_license_in_packages(processed_package=new_package)
+
+                    processed_package = self.processed_packages.get(new_package)
+
+                    if not processed_package:
+                        print(
+                            f"[-] Nothing has been found for package '{real_package_name}' in {self.scanned_json_file}",
+                        )
+                        continue
+
+                    if (
+                        "package_name" not in processed_package
+                        or "version" not in processed_package
```

Comment:
not sure we should skip if missing `version`

Reply:
i checked here because I'm accessing those keys in the `write_to_file` function and it would run into a `KeyError`, and I didn't want to check in the `write_to_file` function. 

I'll refactor, and maybe add the keys with empty strings as values?